src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/PasteOrdering.cs

```diff
@@ -67,6 +67,7 @@ private IPasteDataObjectProcessor PasteProcessor
 
         public bool CanHandleDataObject(object dataObject, IProjectTree dropTarget, IProjectTreeProvider currentProvider)
         {
+            _dropTarget = dropTarget;
```

Comment:
The change looks fine, we can do this in a followup but why do we store it even when we don't handle it? Also, when do we clear the field? We could leak a bunch of data here from another app, right?

Reply:
I did some testing and each F# project gets an instance of "PasteOrdering". So they are isolated and no chance of leaks, AFAIK.

Hmm. After looking at this a bit more, I think I know how to make this make a bit more sense. We should clear the field after we are finished with it and I think after calling the `base.CanHandleDataObject(`, if that is true, I will store the `dropTarget`.

Reply:
Ah I misread this, I thought you were storing "dataObject".

Reply:
Seems I can't do what I wish. So I'll answer your question regarding why we store the drop target even if we don't handle it.

The reason why we have to store the target even if we don't actually paste the item is because there is no other way. We relied on `ProcessDataObjectAsync` to give us the `dropTarget` which seems to have worked in most cases except for what the original issue suggests. If an item that is outside of any project, if I paste that item into a F# project, `ProcessDataObjectAsync` will not be called; I don't know why. But, `CanHandleDataObject` does get called.

Reply:
We don't clear the field, but I think we should after we use the `_dropTarget`.

Reply:
Yeah we should clear it - it will hold onto data long after the item might be deleted/removed from the project.

Reply:
As a separate bug though, it has nothing to do this with this change.

Reply:
I just tried clearing the field after we use the drop target, but `CanHandleDataObject` gets called again after "PasteItemsAsync" and the drop target gets set again, so clearing is useless.